python/pylibcudf/pylibcudf/tests/test_transform.py

```diff
@@ -81,3 +83,33 @@ def test_one_hot_encode():
         schema=pa.schema([pa.field("", pa.bool_(), nullable=False)] * 3),
     )
     assert result.equals(expected)
+
+
+def test_transform_udf():
+    @cuda.jit(device=True)
+    def op(a, b, c):
+        return (a + b) * c
+
+    ptx, _ = cuda.compile_ptx_for_current_device(
+        op, (numba.float64, numba.float64, numba.float64), device=True
+    )
+
+    A = 5.0
+    B = 20.0
+    C = 0.5
+
+    pa_a = pa.array([A] * 100)
+    pa_b = pa.array([B] * 100)
+    pa_c = pa.array([C])
+    pa_expected = pa.array([(A + B) * C] * 100)
+    plc_a = plc.interop.from_arrow(pa_a)
+    plc_b = plc.interop.from_arrow(pa_b)
+    plc_c = plc.interop.from_arrow(pa_c)
+    plc_result = plc.transform.transform(
+        [plc_a, plc_b, plc_c],
+        transform_udf=ptx,
+        output_type=plc.DataType(plc.TypeId.FLOAT64),
+        is_ptx=True,
+    )
+    result = plc.interop.to_arrow(plc_result)
+    assert result.chunk(0).equals(pa_expected)
```

Comment:
Does some thing like this work? You might have to edit `expect`.
```suggestion
    a = pa.array([A] * 100)
    b = pa.array([B] * 100)
    c = pa.array([C])
    expect = pa.array([(A + B) * C] * 100)
    result = plc.transform.transform(
        [
            plc.interop.from_arrow(a),
            plc.interop.from_arrow(b),
            plc.interop.from_arrow(c),
        ],
        transform_udf=ptx,
        output_type=plc.DataType(plc.TypeId.FLOAT64),
        is_ptx=True,
    )
    assert_column_eq(expect, got)
```

Reply:
I used the naming and style convention in the test case above it, as it also made it easier to read (especially the conversions)

Reply:
The extra definitions seem superfluous to me. But no worries this can be improved in subsequent plc PRs

Reply:
fixed, thanks!